pkg/query-service/dao/postgres/connection.go

```diff
@@ -0,0 +1,178 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+	"github.com/pkg/errors"
+	"go.signoz.io/query-service/config"
+	"go.signoz.io/query-service/constants"
+	"go.signoz.io/query-service/model"
+	"go.signoz.io/query-service/telemetry"
+	"go.uber.org/zap"
+)
+
+type modelDao struct {
+	db *sqlx.DB
+}
+
+// InitConn initializes postgres db connection
+func InitConn(pgconf *config.PGConfig) (*sqlx.DB, error) {
+
+	psqlInfo := fmt.Sprintf("dbname=%s user=%s host=%s port=%d ", pgconf.DBname, pgconf.User, pgconf.Host, pgconf.Port)
+	if pgconf.Password != "" {
+		psqlInfo = fmt.Sprintf("%s password=%s", psqlInfo, pgconf.Password)
+	}
+
+	if pgconf.SSLmode != "" {
+		psqlInfo = fmt.Sprintf("%s sslmode=%s", psqlInfo, pgconf.SSLmode)
+	} else {
+		psqlInfo = fmt.Sprintf("%s sslmode=disable", psqlInfo)
+	}
+
+	db, err := sqlx.Open("postgres", psqlInfo)
+	if err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+// InitDB sets up data model (creates tables if necessary) and
+// the connection pool global variable.
+func InitDB(db *sqlx.DB) (*modelDao, error) {
+
+	table_schema := `
+
+		CREATE TABLE IF NOT EXISTS organizations (
```

Comment:
Same question as above, is there a schema or data types b/w sqlite and pg? Any reason not to use one InitDB?

Reply:
The handling of JSON, UUID and SERIALs is different. There is also diff in handling connection pool in sql*lite. It will be good idea to keep the DDLs separate to support future use cases as well. And at some point, I am assuming we will drop sql*lite support.